torchao/quantization/quantize_linear.py

```diff
@@ -0,0 +1,469 @@
+import torch
+import torch.nn as nn
+import logging
+from torchao.quantization.quant_primitives import MappingType
+import torch.nn.functional as F
+from torchao.quantization.unified import Quantizer
+from typing import Optional, Callable, Type, Dict, Any
+from torchao.quantization.utils import (
+    per_token_dynamic_quant,
+    groupwise_affine_quantize_tensor,
+    group_quantize_tensor_symmetric,
+)
+
+
```

Comment:
maybe add a comment to say everything in this file is intended to be deprecated and eventually removed? Then we should also add the new alternative of running these using the quantize_ API, e.g.
```
from torchao.quantization import (
    quantize_,
    int4_weight_only,
    int8_dynamic_activation_int4_weight,
)
quantize_(model, int4_weight_only())
quantize_(model, int8_dynamic_activation_int4_weight())
```

Reply:
Hey, can you elaborate on this step a little bit more? I'm a bit confused.

Reply:
Yeah sure, I mean we can express everything in this file using the new `quantize_` API:

```
# Before:
quantizer = Int4WeightOnlyQuantizer()  # or Int8DynActInt4WeightQuantizer
model = quantizer.quantize(model)

# After:
quantize_(model, int4_weight_only())  # or int8_dynamic_activation_int4_weight
```

We want to deprecate the Quantizer APIs eventually in favor of the quantize_ API, so we want to discourage people from using the quantizers in this file. I think adding a comment here explaining this context will help with that.

Reply:
Thank you, I added the comments to the top of the file noting the deprecation.